src/form/Select/SelectMenu/SelectMenu.tsx

```diff
@@ -210,21 +219,27 @@ export const SelectMenu: FC<SelectMenuProps> = ({
       }}
     >
       <List>
-        {options?.length === 0 && createable && trimmedText && !loading && (
-          <ListItem
-            className="select-menu-create-option"
-            onClick={event => {
-              event.preventDefault();
-              event.stopPropagation();
-              onSelectedChange({
-                value: trimmedText.toLowerCase(),
-                children: trimmedText.toLowerCase()
-              });
-            }}
-          >
-            Create option &quot;{trimmedText.toLowerCase()}&quot;
-          </ListItem>
-        )}
+        {options?.length === 0 &&
+          createable &&
+          trimmedText &&
+          !loading &&
+          (renderCreateOption ? (
+            renderCreateOption(trimmedText, onSelectedChange)
+          ) : (
+            <ListItem
+              className="select-menu-create-option"
+              onClick={event => {
+                event.preventDefault();
+                event.stopPropagation();
+                onSelectedChange({
+                  value: trimmedText.toLowerCase(),
```

Comment:
Wouldn't you want to do this on the `renderCreateOption` callback as well?

Reply:
Not sure how. This object should be prepared on the client side. Like in story